viewing_party/party.py

```diff
@@ -1,23 +1,121 @@
 # ------------- WAVE 1 --------------------
 
+# from lib2to3.pgen2.pgen import generate_grammar
+# from shutil import move
+# from turtle import title
+from collections import Counter
+
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+
+    if title and genre and rating:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+    else: return None
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum_of_ratings = 0
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            sum_of_ratings += movie["rating"]
+    else: return 0
+
+    average_of_ratings = sum_of_ratings / len(user_data["watched"])
+    return average_of_ratings
 
 
+def get_most_watched_genre(user_data):
+    # create a list of all genres to pass into Counter
+    if user_data["watched"]:
+        genre_dict = Counter([movie["genre"] for movie in user_data["watched"]])
+        return genre_dict.most_common(1)[0][0]
+    else: return None
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+# create a helper function to return two lists: my_list_of_movies and my_friends_list_movies
+def get_my_and_friends_movies(user_data):
+    # create a list of my watched movies
+    my_watched_movies = [movie for movie in user_data["watched"]]

```

Comment:
Consider if it's necessary to copy the movies from `user_data["watched"]` into a new list. Now that we have talked about Big(O), we know that this adds a linear operation to both your time and space complexity. Since we already have access to a list of movies the user has watched in the provided data, couldn't we just use `user_data["watched"]` instead?